### PR TITLE
[Android] Fix some System.Net tests

### LINKED ISSUE #37087: System.Net.Requests.Tests tests fail on Android
## System.Net.Requests.Tests
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.SimpleScenario_UseGETVerb_Success(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.GetResponseAsync_UseDefaultCredentials_ExpectSuccess(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.GetResponseAsync_PostRequestStream_ContainsData(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.ResponseUri_GetResponseAsync_ExpectSameUri(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.Headers_GetResponseHeaders_ContainsExpectedValue(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.HaveResponse_GetResponseAsync_ExpectTrue(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.ContentType_AddHeaderWithNoContent_SendRequest_HeaderGetsSent(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.GetResponseAsync_GetResponseStream_ContainsHost(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.SimpleScenario_UsePOSTVerb_Success(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.ContentLength_Get_ExpectSameAsGetResponseStream(useSsl: True)
- [ ] System.Net.Tests.SyncFileWebRequestTestBase.RequestAfterResponse_throws
- [ ] System.Net.Tests.SyncFileWebRequestTestBase.WriteThenReadFile_WriteAccessResultsInNullResponseStream
- [ ] System.Net.Tests.SyncFileWebRequestTestBase.ReadFile_ContainsExpectedContent
- [ ] System.Net.Tests.SyncFileWebRequestTestBase.WriteFile_ContainsExpectedContent
- [ ] System.Net.Tests.HttpWebRequestTest_Async.GetResponseAsync_PostRequestStream_ContainsData(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Async.HaveResponse_GetResponseAsync_ExpectTrue(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Async.ResponseUri_GetResponseAsync_ExpectSameUri(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Async.GetResponseAsync_UseDefaultCredentials_ExpectSuccess(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Async.GetResponseAsync_GetResponseStream_ContainsHost(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Async.SimpleScenario_UseGETVerb_Success(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Async.Headers_GetResponseHeaders_ContainsExpectedValue(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Async.ContentType_AddHeaderWithNoContent_SendRequest_HeaderGetsSent(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Async.ContentLength_Get_ExpectSameAsGetResponseStream(useSsl: True)
- [ ] System.Net.Tests.HttpWebRequestTest_Async.SimpleScenario_UsePOSTVerb_Success(useSsl: True)
- [ ] System.Net.Tests.LoggingTest.EventSource_ExistsWithCorrectId
- [ ] System.Net.Tests.HttpWebRequestTest_Async.HttpWebRequest_Serialize_Fails
- [ ] System.Net.Tests.HttpWebResponseHeaderTest.HttpWebResponse_Serialize_ExpectedResult
- [ ] System.Net.Tests.HttpWebRequestTest_Sync.HttpWebRequest_Serialize_Fails

Edit: Updated based on https://github.com/dotnet/runtime/pull/50095

### LINKED ISSUE #69870: [Android] Some System.Net.Http.Functional.Tests failing
Some functional tests are failing on Android:

System.Net.Http.Functional.Tests.HPackTest
    - HPack_HeaderEncoding

System.Net.Http.Functional.Tests.Http1CloseResponseStreamZeroByteReadTest
    - ZeroByteRead_IssuesZeroByteReadOnUnderlyingStream

System.Net.Http.Functional.Tests.Http1MultiChunkResponseStreamZeroByteReadTest
    - ZeroByteRead_IssuesZeroByteReadOnUnderlyingStream

System.Net.Http.Functional.Tests.Http1RawResponseStreamZeroByteReadTest
    - ZeroByteRead_IssuesZeroByteReadOnUnderlyingStream

System.Net.Http.Functional.Tests.Http1SingleChunkResponseStreamZeroByteReadTest
    - ZeroByteRead_IssuesZeroByteReadOnUnderlyingStream

System.Net.Http.Functional.Tests.Http2ResponseStreamZeroByteReadTest
    - ZeroByteRead_BlocksUntilDataIsAvailable

System.Net.Http.Functional.Tests.HttpClientTest
    - SendAsync_CorrectVersionSelected_LoopbackServer

System.Net.Http.Functional.Tests.SocketsHttpHandler_DiagnosticsTest_Http2
    - SendAsync_CustomSocketsHttpHandlerPropagator_PropagatorIsUsed
    - SendAsync_HeadersAreInjectedOnRedirects

System.Net.Http.Functional.Tests.SocketsHttpHandler_Http2_TrailingHeaders_Test
    - Http2GetAsync_MissingTrailer_TrailingHeadersAccepted
    - Http2GetAsync_NoTrailingHeaders_EmptyCollection
    - Http2GetAsync_TrailerHeaders_TrailingHeaderNoBody
    - Http2GetAsync_TrailerHeaders_TrailingPseudoHeadersThrow
    - Http2GetAsync_TrailingHeaders_NoData_EmptyResponseObserved
    - Http2GetAsyncResponseHeadersReadOption_TrailingHeaders_Available

System.Net.Http.Functional.Tests.SocketsHttpHandler_Http2FlowControl_Test
    - BadRttPingResponse_RequestShouldFail
    - InitialHttp2StreamWindowSize_SentInSettingsFrame

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_Cancellation_Test_Http2
    - GetAsync_CancelDuringResponseBodyReceived_Unbuffered_TaskCanceledQuickly
    - GetAsync_CancelDuringResponseHeadersReceived_TaskCanceledQuickly
    - GetAsync_CancelPendingRequests_DoesntCancelReadAsyncOnResponseStream
    - PostAsync_CancelDuringRequestContentSend_TaskCanceledQuickly

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_ClientCertificates_Test
    - Manual_CertificateOnlySentWhenValid_Success

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_ConnectionPooling_Test
    - Http2_SmallConnectionTimeout_SubsequentRequestUsesDifferentConnection

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_DangerousAcceptAllCertificatesValidator_Test
    - SetDelegate_ConnectionSucceeds

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_Finalization_Http2_Test
    - IncompleteResponseStream_ResponseDropped_CancelsRequestToServer

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_Proxy_Test
    - ProxyTunnelRequest_GetAsync_Success
    - ProxyTunnelRequest_MaxConnectionsSetButDoesNotApplyToProxyConnect_Success
    - ProxyTunnelRequest_OriginServerSendsProxyAuthChallenge_NoProxyAuthPerformed

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_ServerCertificates_Test
    - UseCallback_SelfSignedCertificate_ExpectedPolicyErrors

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_SslProtocols_Test
    - GetAsync_AllowedSSLVersion_Succeeds
    - GetAsync_NoSpecifiedProtocol_DefaultsToTls12

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandlerTest
    - GetAsync_IPBasedUri_Success
    - GetAsync_IPv6LinkLocalAddressUri_Success
    - GetAsync_SecureAndNonSecureIPBasedUri_CorrectlyFormatted

System.Net.Http.Functional.Tests.SocketsHttpHandler_RequestContentLengthMismatchTest_Http2
    - ContentLength_DoesNotMatchRequestContentLength_Throws

System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_ConnectCallback_Http11
    - ConnectCallback_BindLocalAddress_Success
    - ConnectCallback_ConnectionPrefix_Success
    - ConnectCallback_DerivedSslStream_OK
    - ConnectCallback_NoAlpn_OK
    - ConnectCallback_SslStream_OK
    - ConnectCallback_UseMemoryBuffer_Success
    - ConnectCallback_UseUnixDomainSocket_Success

System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_ConnectCallback_Http2
    - ConnectCallback_BindLocalAddress_Success
    - ConnectCallback_ConnectionPrefix_Success
    - ConnectCallback_ContextHasCorrectProperties_Success
    - ConnectCallback_DerivedSslStream_OK
    - ConnectCallback_NoAlpn_OK
    - ConnectCallback_SslStream_OK
    - ConnectCallback_UseMemoryBuffer_Success
    - ConnectCallback_UseUnixDomainSocket_Success

System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Cookies_Http2
    - GetAsync_AddCookieHeader_CookieHeaderSent
    - GetAsync_AddMultipleCookieHeaders_CookiesSent
    - GetAsync_CookiePathDoesNotMatchRequestPath_CookieAccepted
    - GetAsync_DefaultCoookieContainer_NoCookieSent
    - GetAsync_NoPathDefined_CookieAddedWithDefaultPath
    - GetAsync_ReceiveInvalidSetCookieHeader_ValidCookiesAdded
    - GetAsync_ReceiveMultipleSetCookieHeaders_CookieAdded
    - GetAsync_ReceiveSetCookieHeader_CookieAdded
    - GetAsync_ReceiveSetCookieHeader_CookieRemoved
    - GetAsync_ReceiveSetCookieHeader_CookieUpdated
    - GetAsync_Redirect_CookiesArePreserved
    - GetAsync_SetCookieContainer_CookieSent
    - GetAsync_SetCookieContainerAndCookieHeader_BothCookiesSent
    - GetAsync_SetCookieContainerAndMultipleCookieHeaders_BothCookiesSent
    - GetAsync_SetCookieContainerMultipleCookies_CookiesSent
    - GetAsyncWithBasicAuth_ReceiveSetCookie_CookieSent
    - GetAsyncWithRedirect_ReceiveSetCookie_CookieSent
    - GetAsyncWithRedirect_SetCookieContainer_CorrectCookiesSent

System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2
    - CompletedResponse_FrameReceived_Ignored
    - CompletedResponse_WindowUpdateFrameReceived_Success
    - DataFrame_IdleStream_ConnectionError
    - DataFrame_NoStream_ConnectionError
    - DataFrame_TooLong_ConnectionError
    - DynamicTable_Receipt_Success
    - DynamicTable_Resize_Success
    - DynamicTable_Reuse_Concat
    - DynamicTableSizeUpdate_Exceeds_Settings_Throws
    - EmptyResponse_FrameReceived_Ignored
    - GetAsync_StreamRefused_RequestIsRetried
    - GoAwayFrame_AbortAllPendingStreams_StreamFailWithExpectedException
    - GoAwayFrame_AllPendingStreamsValid_RequestsSucceedAndConnectionClosed
    - GoAwayFrame_NewRequest_NewConnection
    - GoAwayFrame_NonzeroStream_ConnectionError
    - GoAwayFrame_NoPendingStreams_ConnectionClosed
    - GoAwayFrame_RequestInFlight_Finished
    - GoAwayFrame_RequestWithBody_ServerDisconnect_AbortStreamsAndThrowIOException
    - GoAwayFrame_ServerDisconnect_AbortStreams
    - GoAwayFrame_UnprocessedStreamFirstRequestFinishedFirst_RequestRestarted
    - GoAwayFrame_UnprocessedStreamFirstRequestWaitsUntilSecondFinishes_RequestRestarted
    - HeadersFrame_IdleStream_ConnectionError
    - Http2_ClientPreface_Sent
    - Http2_DataSentBeforeServerPreface_ProtocolError
    - Http2_InitialSettings_SentAndAcked
    - Http2_MultipleConnectionsEnabled_ManyRequestsEnqueuedSimultaneously_SufficientConnectionsCreated
    - Http2_MultipleConnectionsEnabled_OpenAndCloseMultipleConnections_Success
    - Http2_NoResponseBody_Success
    - Http2_PendingReceive_SendsReset
    - Http2_PendingSend_Cancellation
    - Http2_ProtocolMismatch_Throws
    - Http2_SendOverConnectionWindowSizeWithoutExplicitFlush_ClientSendsUpToFullWindowSize
    - Http2_SendOverStreamWindowSizeWithoutExplicitFlush_ClientSendsUpToFullWindowSize
    - Http2_ServerSendsInvalidSettingsValue_Error
    - Http2_ServerSendsValidSettingsValues_Success
    - Http2_StreamResetByServerAfterHeadersSent_RequestFails
    - Http2_StreamResetByServerAfterPartialBodySent_RequestFails
    - Http2_StreamResetByServerBeforeHeadersSent_RequestFails
    - Http2_WaitingOnWindowCredit_Cancellation
    - Http2_ZeroLengthResponseBody_Success
    - Http2Connection_Should_Not_Wrap_HttpContent_CustomException
    - Http2Connection_Should_Wrap_HttpContent_InvalidOperationException
    - Http2GetAsync_MultipleStatusHeaders_Throws
    - Http2GetAsync_StatusHeaderNotFirst_Throws
    - Http2GetAsync_TrailigPseudo_Throw
    - Http2SendAsync_LargeHeaders_CorrectlyWritten
    - InboundWindowSize_Exceeded_Throw
    - MaxResponseHeadersLength_Exact_Success
    - MaxResponseHeadersLength_Exceeded_Throws
    - MaxResponseHeadersLength_Malicious_Throws
    - PostAsync_StreamRefused_RequestIsRetried
    - PostAsyncDuplex_ClientSendsEndStream_Success
    - PostAsyncDuplex_RequestContentException_ResetsStream
    - PostAsyncDuplex_RequestContentExceptionAfterResponseEndReceivedButBeforeConsumed_ResetsStreamAndThrowsOnResponseStreamRead
    - PostAsyncDuplex_ServerCompletesResponseBodyThenResetsStreamWithNoError_SuccessAndRequestBodyCancelled
    - PostAsyncDuplex_ServerResetsStream_Throws
    - PostAsyncDuplex_ServerSendsEndStream_Success
    - PostAsyncExpect100Continue_NonSuccessResponse_RequestBodyNotSent
    - PostAsyncExpect100Continue_SendRequest_Ok
    - PostAsyncNonDuplex_ServerCompletesResponseBodyThenResetsStreamWithNoError_SuccessAndRequestBodyCancelled
    - ResetResponseStream_FrameReceived_Ignored
    - ResponseStreamFrames_ContinuationBeforeHeaders_ConnectionError
    - ResponseStreamFrames_DataAfterHeadersAndContinuationWithoutEndHeaders_ConnectionError
    - ResponseStreamFrames_DataAfterHeadersWithoutEndHeaders_ConnectionError
    - ResponseStreamFrames_DataBeforeHeaders_ConnectionError
    - ResponseStreamFrames_HeadersAfterHeadersAndContinuationWithoutEndHeaders_ConnectionError
    - ResponseStreamFrames_HeadersAfterHeadersWithoutEndHeaders_ConnectionError

System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_HttpClientHandlerTest_Headers_Http2
    - GetAsync_EmptyResponseHeader_Success
    - GetAsync_LargeHeader_Success
    - GetAsync_MissingExpires_ReturnNull
    - SendAsync_CustomRequestEncodingSelector_CanSendNonAsciiHeaderValues
    - SendAsync_CustomResponseEncodingSelector_CanReceiveNonAsciiHeaderValues
    - SendAsync_DefaultHeaders_CorrectlyWritten
    - SendAsync_Expires_Success
    - SendAsync_LargeHeaders_CorrectlyWritten
    - SendAsync_RequestHeaderInResponse_Success
    - SendAsync_RequestWithSimpleHeader_ResponseReferencesUnmodifiedRequestHeaders
    - SendAsync_SpecialHeaderKeyOrValue_Success
    - SendAsync_UserAgent_CorrectlyWritten
    - SendAsync_WithZeroLengthHeaderName_Throws

System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_HttpClientHandlerTest_Http2
    - Dispose_DisposingHandlerCancelsActiveOperationsWithoutResponses
    - GetAsync_ExpectContinueTrue_NoContent_StillSendsHeader
    - GetAsync_IPBasedUri_Success
    - GetAsync_IPv6LinkLocalAddressUri_Success
    - GetAsync_ServerNeedsNonStandardAuthAndSetCredential_StatusCodeUnauthorized
    - PostAsync_ManyDifferentRequestHeaders_SentCorrectly
    - ReadAsStreamAsync_EmptyResponseBody_HandlerProducesWellBehavedResponseStream
    - ReadAsStreamAsync_HandlerProducesWellBehavedResponseStream
    - SendAsync_1xxResponsesWithHeaders_InterimResponsesHeadersIgnored
    - SendAsync_Expect100Continue_RequestBodyFails_ThrowsContentException
    - SendAsync_MultipleExpected100Responses_ReceivesCorrectResponse
    - SendAsync_No100ContinueReceived_RequestBodySentEventually
    - SendAsync_RequestVersion20_HttpNotHttps_NoUpgradeRequest
    - SendAsync_Unexpected1xxResponses_DropAllInterimResponses
    - SendRequest_UriPathHasReservedChars_ServerReceivedExpectedPath

System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_PlaintextStreamFilter_Http11
    - PlaintextStreamFilter_ConnectionPrefix_Success
    - PlaintextStreamFilter_CustomStream_Success
    - PlaintextStreamFilter_ExceptionDuringCallback_ThrowsHttpRequestExceptionWithInnerException
    - PlaintextStreamFilter_Logging_Success
    - PlaintextStreamFilter_SimpleDelegatingStream_Success

System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_PlaintextStreamFilter_Http2
    - PlaintextStreamFilter_ConnectionPrefix_Success
    - PlaintextStreamFilter_ExceptionDuringCallback_ThrowsHttpRequestExceptionWithInnerException
    - PlaintextStreamFilter_SimpleDelegatingStream_Success

System.Net.Http.Functional.Tests.SocksProxyTest_Http1_Async
    - TestLoopbackAsync

System.Net.Http.Functional.Tests.SocksProxyTest_Http2
    - TestLoopbackAsync

System.Net.Http.Functional.Tests.Http1ContentLengthResponseStreamZeroByteReadTest

**Failing outer loop tests:**

System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2
    - PostAsyncDuplex_DisposeResponseBodyBeforeEnd_ResetsStreamAndThrowsOnRequestStreamWriteAndResponseStreamRead
    - SendAsync_ConcurentSendReceive_Fail
    - GetAsync_SettingsFrameNotSentOnNewConnection_ClientApplies100StreamLimit
    - PostAsyncDuplex_DisposeResponseBodyAfterEndReceivedButBeforeConsumed_ResetsStreamAndThrowsOnRequestStreamWriteAndResponseStreamRead
    - SendAsync_StreamContentRequestBody_WaitsForRequestBodyToComplete
    - SendAsync_ConcurentSendReceive_Ok
    - PostAsyncDuplex_CancelledBeforeResponseHeadersReceived_ResetsStream
    - Http2_PendingSend_SendsReset
    - Http2_FlowControl_ClientDoesNotExceedWindows
    - Http2_MultipleConnectionsEnabled_IdleConnectionTimeoutExpired_ConnectionRemovedAndNewCreated
    - SocketSendQueueFull_RequestCanceled_ThrowsOperationCanceled
    - GetAsync_ServerDelaysSendingSettingsThenSetsLowerMaxConcurrentStreamsLimitThenIncreaseIt_ClientAppliesEachLimitChangeProperly
    - Http2_InitialWindowSize_ClientDoesNotExceedWindows
    - GetAsync_ServerSendSettingsWithoutMaxConcurrentStreams_ClientAppliesInfiniteLimit
    - PostAsyncDuplex_FinishRequestBodyAndDisposeResponseBodyAfterEndReceivedButBeforeConsumed_DoesNotResetStream
    - Http2_WaitingForStream_Cancellation
    - Http2_MaxConcurrentStreams_LimitEnforced

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_ServerCertificates_Test
    - UseCallback_BadCertificate_ExpectedPolicyErrors

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_Http11_Cancellation_Test
    - ConnectionFailure_AfterInitialRequestCancelled_SecondRequestSucceedsOnNewConnection

System.Net.Http.Functional.Tests.SocketsHttpHandler_Http2FlowControl_Test
    - MaxStreamWindowSize_WhenSet_WindowDoesNotScaleAboveMaximum
    - StreamWindowScaleThresholdMultiplier_LowValue_WindowScalesFaster
    - HighBandwidthDelayProduct_ClientStreamReceiveWindowWindowScalesUp
    - StreamWindowScaleThresholdMultiplier_HighValue_WindowScalesSlower
    - DisableDynamicWindowScaling_HighBandwidthDelayProduct_WindowRemainsConstant

System.Net.Http.Functional.Tests.SocketsHttpHandler_Http2KeepAlivePing_Test
    - KeepAliveConfigured_KeepAlivePingsAreSentAccordingToPolicy
    - KeepAlivePingDelay_Infinite_NoKeepAlivePingIsSent
    - KeepAliveConfigured_NoPingResponseDuringActiveStream_RequestShouldFail
    - HttpKeepAlivePingPolicy_Always_NoPingResponseBetweenStreams_SecondRequestShouldFail

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_DangerousAcceptAllCertificatesValidator_Test
    - InvalidCertificateServers_CertificateValidationDisabled_Succeeds

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_Cancellation_Test_Http2
    - Expect100Continue_WaitsExpectedPeriodOfTimeBeforeSendingContent
    - ConnectionFailure_AfterInitialRequestCancelled_SecondRequestSucceedsOnNewConnection

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_ConnectionPooling_Test
    - HandlerDroppedWithoutDisposal_NotKeptAlive

System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_ClientCertificates_Test
    - Manual_CertificateSentMatchesCertificateReceived_Success

### PR DESCRIPTION
**Pull request by simonrozsival**
We use some self-signed certificates in our tests and Android doesn't allow traffic to the servers using those certificates. In order to get the tests to work, we need to configure include the public keys in the .apk we build and use [Network Security Configuration](https://developer.android.com/training/articles/security-config) to trust our self-signed certificates.

Closes #37087
Fixes most of #68206
Closes #69870
Fixes Android-related failures in #72209

### CODE REVIEW

**Comment by simonrozsival** [comment]
The tests pass on Android devices (arm and arm64 lanes) and fail on emulators (x86 and x64 lanes). I'm converting this PR to a draft until I figure out why.

File: src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.ServerCertificates.cs [hunk 1]
```diff
@@ -282,7 +282,7 @@ private async Task UseCallback_BadCertificate_ExpectedPolicyErrors_Helper(string
         [OuterLoop("Uses external servers")]
         [Theory]
         [MemberData(nameof(CertificateValidationServersAndExpectedPolicies))]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/69870", TestPlatforms.Android)]
+        [SkipOnPlatform(TestPlatforms.Android, "Android rejects the certificate")]
```
Comment (akoeplinger) [inline src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.ServerCertificates.cs:285]: same comment about the reason

**Review by akoeplinger — APPROVED** [review]
Looks good apart from a few comments and what we discussed privately :)

File: src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs [hunk 2]
```diff
@@ -62,20 +59,29 @@ public async Task ServerAsyncAuthenticate_EachSupportedProtocol_Success(SslProto
                 });
 
             Assert.NotNull(e);
-            Assert.IsAssignableFrom(expectedException, e);
+
+            if (PlatformDetection.IsAndroid)
+            {
+                Assert.True(
+                    e is AuthenticationException || e is System.IO.IOException,
+                    $"the exception should be either AuthenticationException or IOException and not {e.GetType().FullName}"
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs]: I decided to keep the test disabled

**Pull request merged**